schemas/v3.0/schema.yaml

```diff
@@ -254,31 +255,57 @@ definitions:
       type:
         type: string
         enum:
-         - array
-         - boolean
-         - integer
-         - number
-         - object
-         - string
+          - array
+          - boolean
+          - integer
+          - number
+          - object
+          - string
       not:
-        $ref: '#/definitions/Schema'
+        oneOf:
+          - $ref: '#/definitions/Schema'
+          - $ref: '#/definitions/Reference'
+      allOf:
+        type: array
+        items:
+          oneOf:
+            - $ref: '#/definitions/Schema'
+            - $ref: '#/definitions/Reference'
+      oneOf:
+        type: array
+        items:
+          oneOf:
+            - $ref: '#/definitions/Schema'
+            - $ref: '#/definitions/Reference'
+      anyOf:
+        type: array
+        items:
+          oneOf:
+            - $ref: '#/definitions/Schema'
+            - $ref: '#/definitions/Reference'
+      items:
+        type: array
```

Comment:
Wondering why array's item type is `array` and not an `object`. From the [OAS3 spec](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#properties):
> items - Value MUST be an object and not an array. Inline or referenced schema MUST be of a Schema Object and not a standard JSON Schema. <...>